etl-pipeline/processes/record_candidate_finder.py

```diff
@@ -0,0 +1,194 @@
+import re
+from logging import Logger
+from typing import List, Set, Tuple, Optional, Any
+from sqlalchemy.exc import DataError
+from managers import DBManager
+
+
+from model import Record
+from logger import create_log
+
+logger = create_log(__name__)
+
+
+class CandidateRecordFinder:
+    
+    # Maximum number of "hops" to follow when matching records through identifiers
+    MAX_MATCH_DISTANCE = 4
```

Comment:
Is there a better variable name that explains what we are doing here with the hops?